### PR TITLE
Add layout customisation support for `SongProgress`

### PR DESCRIPTION
**Pull request by peppy**
Eventually need to implement the legacy version of this...

### CODE REVIEW

File: osu.Game/Skinning/Editor/SkinComponentToolbox.cs [hunk 1]
```diff
@@ -92,6 +92,9 @@ private static ToolboxComponentButton attemptAddComponent(Type type)
 
         private class ToolboxComponentButton : OsuButton
         {
+            protected override bool ReceivePositionalInputAtSubTree(Vector2 screenSpacePos) => false;
+            public override bool PropagateNonPositionalInputSubTree => false;
```
Comment (smoogipoo) [inline osu.Game/Skinning/Editor/SkinComponentToolbox.cs:96]: I assume this is just to stop children from receiving input? Does:
`protected override bool ShouldBeConsideredForInput(Drawable child)` work?

Comment (peppy) [reply osu.Game/Skinning/Editor/SkinComponentToolbox.cs:96]: It does work. Is that preferred?

Comment (smoogipoo) [reply osu.Game/Skinning/Editor/SkinComponentToolbox.cs:96]: It's used in other cases, so I think so.

For me, I find this hard to read as is - the `ReceivePositionalInputAtSubTree` line makes it seem like it should stop `OnClick()` input so I don't know why it's working at all.

Comment (peppy) [reply osu.Game/Skinning/Editor/SkinComponentToolbox.cs:96]: As mentioned on the framework side, there's a disconnect between `ReceivePositionalInputAtSubTree` and `PropagatePositionalInputSubTree` which probably needs to be resolved..

Comment (smoogipoo) [reply osu.Game/Skinning/Editor/SkinComponentToolbox.cs:96]: Ah, I was glossing over the `SubTree` suffix... Still prefer using `ShouldBeConsideredForInput`...

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**